src/middlewares/DataEngine.js

```diff
@@ -0,0 +1,42 @@
+import Customize from '../helpers/Customize';
+/**
+ * @export
+ * @class DataEngine
+ */
+class DataEngine {
+  /**
+   * User can be able to fetch a single row from a table
+   * @static
+   * @param {object} req request object
+   * @param {object} res response object
+   * @param {object} table table
+   * @param {object} condition bject whose key is the column and value is the value
+   * @memberof UserController
+   * @returns {object} either an error or calls next
+   */
+  static async findOne(req, res, table, condition) {
+    const oneRow = await table.findOne({ where: condition });
+    if (!oneRow) {
+      return Customize.errorMessage(req, res, 'Not found', 404);
+    }
+    req.row = oneRow.dataValues;
+  }
+
+  /**
+   * User can be able to fetch a single row from a table
+   * @static
+   * @param {object} req request object
+   * @param {object} res response object
+   * @param {object} table table
+   * @param {object} condition bject whose key is the column and value is the value
+   * @memberof UserController
+   * @returns {object} either an error or calls next
+   */
+  static async findConflict(req, res, table, condition) {
+    const oneRow = await table.findOne({ where: condition });
+    if (oneRow) {
+      return Customize.errorMessage(req, res, 'Conflict', 409);
+    }
```

Comment:
I think you can specify the conflict. For instance, by passing the conflict message in `find conflict` parameter